specs/EnhancedSecurityMode.md

```diff
@@ -0,0 +1,468 @@
+  /// the value of PreferredEnhancedSecurityModeLevel. However, this is not respected if 
+  /// ICoreWebView2StagingEnvironmentOptions9::IsEnhancedSecurityModeEnabled is set to false. This means that when you update the list
+  /// when ESM is disabled, it will be updated and persisted, but will not take effect until the feature is enabled. 
+  /// 
+  /// See ICoreWebView2StagingEnvironmentOptions9::IsEnhancedSecurityModeEnabled for more details.
+  /// 
+  /// The uris parameter is a list of uri filters. It can also take wildcard strings and matches against the navigation uri.
+  /// See https://www.chromium.org/administrators/url-blocklist-filter-format/ for more details.
+  /// 
+  /// Some examples of allowed strings are as follows:
+  /// | URI filter in list      | URI to navigate          | ESM applied? |
+  /// |___________________________________________________________________|
+  /// | *://example.com         | www.example.com          | yes          |
+  /// | *://example.com         | www.example2.com         | no           |
+  /// | https://www.example.com | www.example.com/hi       | yes          |
+  /// | https://sample.com      | www.sample.com           | no           |
+  /// | https://*               | https://www.example.com  | yes          |
+  /// | example.com             | https://www.example.com  | yes          |
+  /// | *                       | https://www.example.com  | yes          | 
+  /// 
+  /// 
+  HRESULT GetEnhancedSecurityModeBypassList(
+      [out] UINT32* listCounts,
+      [out] LPWSTR** uriFilters
+  );
+  ///
+  /// Set the array of URIs in the allow list.
+  /// 
+  HRESULT SetEnhancedSecurityModeBypassList(
+      [in] UINT32 listCounts,
+      [in] LPCWSTR* uriFilters
+  );
+  ///
+  /// The EnhancedSecurityModeEnforceList property allows you to add URI filters to the enforce list which are associated
+  /// with a profile. This property would apply to the context of the profile. That is, all
+  /// WebView2s sharing the same profile will be affected. The value is also persisted in the user data folder.
+  /// 
+  /// This means that if a site is in the enforce list, the ESM level for that site will always be set to Strict, regardless of 
+  /// the value of PreferredEnhancedSecurityModeLevel. However, this is not respected if 
+  /// ICoreWebView2StagingEnvironmentOptions9::IsEnhancedSecurityModeEnabled is set to false. This means that when you update the list
+  /// when ESM is disabled, it will be updated and persisted, but will not take effect until the feature is enabled. 
+  /// See ICoreWebView2StagingEnvironmentOptions9::IsEnhancedSecurityModeEnabled for more details.
+  /// 
+  /// The enforce list takes precedence over any other list. That is, if a site URL is both in the enforce and bypass list, its ESM level will
+  /// be set to Strict.
+  /// 
+  /// 
+  HRESULT GetEnhancedSecurityModeEnforceList(
```

Comment:
We would normally use Get/Set methods like this to represent an array. It would probably be easier (at least in .NET / WinRT) to use a vector/collection here instead. That should be the mutable collection pattern (see the WebView2 API Patterns doc). In COM that's get_EnhancedSecurityModeEnforceList([out] ICoreWebView2StringCollection** value) - a getter only that returns a mutable collection of strings. In WinRT/MIDL3 this should be Vector<String> EnhancedSecurityModeEnforceList { get; } and in .NET a string collection property.